components/service/glean/src/main/java/mozilla/components/service/glean/LabeledMetricType.kt

```diff
@@ -106,6 +119,24 @@ data class LabeledMetricType<T>(
         }
     }
 
+    /**
+     * Get the storage engine associated with a given metric type.
+     *
+     * @return A storage engine
+     */
+    internal fun getStorageEngineForMetric(): StorageEngine? {
+        // Every metric that supports labels needs an entry here
+        return when (subMetric) {
```

Comment:
nit: What if we had this kind of functionality as part of the `StorageEngineManager`? It would be a bit less weird as the manager should know about the engines. What do you think?

Reply:
That makes sense.  Easy enough to move it.